libs/sdk-bindings/bindings-android/lib/src/main/kotlin/breez_sdk_notification/ForegroundService.kt

```diff
@@ -46,19 +53,32 @@ abstract class ForegroundService : SdkForegroundService, Service() {
         return null
     }
 
+    override fun onFinished(job: Job) {
+        synchronized(this) {
+            logger.log(TAG, "Job has finished: $job", "DEBUG")
+            jobs.remove(job)
+            delayedShutdown()
+        }
+    }
+
     /** Stop the service */
     private val shutdownHandler = Handler(Looper.getMainLooper())
     private val shutdownRunnable: Runnable = Runnable {
         logger.log(TAG, "Reached scheduled shutdown...", "DEBUG")
         shutdown()
     }
 
-    override fun pushbackShutdown() {
+    private fun resetShutdown() {
         shutdownHandler.removeCallbacksAndMessages(null)
-        shutdownHandler.postDelayed(shutdownRunnable, SHUTDOWN_DELAY_MS)
     }
 
-    override fun shutdown() {
+    private fun delayedShutdown() {
+        if (jobs.isEmpty()) {
+            shutdownHandler.postDelayed(shutdownRunnable, SHUTDOWN_DELAY_MS)
```

Comment:
What is the reason not to shutdown immediately here?

Reply:
I thought maybe it's good to leave the SDK running a bit longer rather than immediately shutting down after the last job. What do you think?

Reply:
I think if the jobs signals then it means it has been completed so I don't see any particular reason right now.
I am OK with adding several seconds, can't hurt, but right now it adds 1 minute.

Reply:
Ok, I will remove the delay then

Reply:
> What is the reason not to shutdown immediately here?
@roeierez The reason is to keep the service running for other tasks that may come while we have an SDK connection.